### PR TITLE
HBase backed ingest/cli tools [WIP]

### PR DESCRIPTION
**Pull request by moradology**
This PR adds command line tooling to (locally) ingest and query against OSM features in a GeoMesa cluster via `scripts/cli.sh`

### CODE REVIEW

**Comment by mojodna** [comment]
@moradology can you drop some knowledge on how to get this running?

I managed to build `src/ingest`, but the JAR names differ (`osmesa-ingest_2.11-0.1.0.jar` vs. `osmesa-ingest.jar`) and when i symlink things, it says

```
$ ./scripts/cli.sh ingest
Template config: /etc/hadoop/conf/core-site.xml from /etc/hadoop/conf/core-site.xml.template
Template config: /etc/hadoop/conf/hdfs-site.xml from /etc/hadoop/conf/hdfs-site.xml.template
Exception in thread "main" java.lang.NoClassDefFoundError: com/monovore/decline/CommandApp
```

so something is amiss in my `CLASSPATH` (or the CLI Docker image is missing something)

**Comment by moradology** [comment]
I should say that this is still WIP, but the jar name is what's messed up there. I was doing some code organization and must've missed that. I'll push up a commit in a few minutes

**Comment by mojodna** [comment]
It looks like this might work (still waiting for it to finish):

```bash
cd src
./sbt "project ingest" assembly
```

**Comment by mojodna** [comment]
(I know practically nothing about `sbt`, so that's part of the reason I'm running into problems, I suspect.)

**Comment by moradology** [comment]
SBT has its problems, but I've come to like how terse it is and its plugin system. To get things running, I'd run (in vagrant):
```bash
./scripts/update.sh
./scripts/services.sh
```
In another session (once you've run assembly for the ingest project):
```bash
./scripts/cli.sh ingest \
  --orc s3://bucket/with/some.orc \
  --conf ingest/src/main/resources/local/datastore.yaml \
  --local
```
The hbase catalog will be 'turkmenistan-roads' — an artifact of the file I was testing this with.

**Comment by mojodna** [comment]
Working!

I had to create an ORC file from the [latest (non-history) Turkmenistan PBF](http://download.geofabrik.de/asia/turkmenistan.html). `suriname.orc` doesn't contain the `visible` column and using history dumps result in NPEs, potentially because lat/lon fields are missing (something I ran into when processing history dumps recently):

```
Caused by: java.lang.NullPointerException
	at vectorpipe.osm.package$$anonfun$4.apply(package.scala:77)
	at vectorpipe.osm.package$$anonfun$4.apply(package.scala:76)
```

**Comment by mojodna** [comment]
See https://github.com/geotrellis/vectorpipe/issues/30 for the vectorpipe issue

File: deployment/hbase_performance.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+## S3-backed HBase performance
```
Comment (lossyrob) [inline deployment/hbase_performance.md]: 👍

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/FootprintByCampaign.scala [hunk 2]
```diff
@@ -1,143 +0,0 @@
-package osmesa.analytics.oneoffs
```
Comment (lossyrob) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/FootprintByCampaign.scala]: On merge, shouldn't delete these, let the upstream reorder things

File: src/query/src/main/scala/osmesa/query/model/Hashtag.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package osmesa.query.model
+
+import io.circe.generic.JsonCodec
+
+import scala.util.Random
+
+
+@JsonCodec
+case class Hashtag(tag: String, count: Int)
```
Comment (lossyrob) [inline src/query/src/main/scala/osmesa/query/model/Hashtag.scala]: Should this be called "UserHashtag"? Seems specific to the user, as opposed to the hashtag for a campaign.

File: src/query/src/main/scala/osmesa/query/model/OsmUser.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import io.circe.generic.JsonCodec
+import cats._
+import cats.implicits._
+
+import scala.util.Random
+import java.time.Instant
+
+
+@JsonCodec
+case class OsmUser(
```
Comment (lossyrob) [inline src/query/src/main/scala/osmesa/query/model/OsmUser.scala]: Rename form OsmUser -> Contributor

**Comment by moradology** [comment]
Closing this PR to pick changes from it for smaller merges against #20

**Pull request closed without merging**